### PR TITLE
JSDoc

### PR DESCRIPTION
**Pull request by rsimon**
# In this PR

This PR adds JSDoc to the code relating to the Store & related utility functions.

### CODE REVIEW

**Comment by rsimon** [comment]
P.S.: In parallel, I also started [writing a proper README fo Peripleo](https://github.com/peripleo/peripleo). I think I'll move some of this (specifically the list of UI components) into Wiki pages later. Either way: do let me know if you have feedback.

File: src/components/ITSBStore/itsbGraph.js [hunk 1]
```diff
@@ -110,8 +130,30 @@ export class ITSBGraph {
...
+   * Checks if one or more nodes with the given ID(s)
+   * exist in the graph. The method accepts multiple arguments.
+   * You can check for a single ID by calling `exists('id1')`,
+   * or for multiple IDs by calling `exists('id1', 'id2', 'id3').
+   * The return value will be `true` if ALL IDs exist in the graph.
+   *
+   * @param {...string} arg one or multiple node ids
+   * @returns {boolean} true if ALL of the given IDs exist
+   */
+  exists() {
```
Comment (blms) [inline src/components/ITSBStore/itsbGraph.js:143]: This is a JS pattern that I rarely see used! Do you prefer this over a ["rest" parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters)? Generally I prefer the latter.

File: src/components/ITSBStore/itsbGraph.js [hunk 2]
```diff
@@ -136,10 +186,46 @@ export class ITSBGraph {
...
+   * date range).
+   *
+   * The response format is an array of itinerary objects:
+   *
+   * {
+   *   'author': 'author-id',
+   *   'waypoints': [
+   *     //...list of waypoint nodes, sorted by time
+   *   ]
+   * }
```
Comment (blms) [inline src/components/ITSBStore/itsbGraph.js:224]: Appreciate the response format example :)

File: src/components/ITSBStore/utils/likelihoodUtils.js [hunk 3]
```diff
@@ -54,6 +69,8 @@ export const estimateInterval = (waypoint, graph) => {
       return { waypoint, start, end, likelihood: 3 };
     }
   } else {
+    // Interval 'inference' (as described in the comment above) happens
+    // here - see method below
```
Comment (blms) [inline src/components/ITSBStore/utils/likelihoodUtils.js:73]: Can we move this comment to JSDoc for `inferInterval`? (something like "Helper method to compute the interval inferred according to the process as outlined in the documentation for `estimateInterval`")

File: src/components/ITSBStore/utils/waypointUtils.js [hunk 4]
```diff
@@ -73,8 +77,20 @@ export const sortWaypointsBySequence = (waypoints, graph) => {
...
+ * Checks if a 'when'-timespan (object with {start} and/or {end})
+ * is in the given date range. The format of the date range is the same
+ * as used in ITSBSearchHandler, i.e. an array of two ISO-formatted strings.
+ *
+ * The method returns true under different conditions:
+ * - If the the timespan has a defined start AND end, the method returns
+ *   true if the timespan is FULLY INCLUDED in the date range. (I.e. it's
+ *   not enough if the timespan only intersects).
+ * - If the timespan has ONLY a start OR end date, the method returns true
+ *   if the defined start or end date is inside the given date range.
```
Comment (blms) [inline src/components/ITSBStore/utils/waypointUtils.js:89]: Great explanation!

File: src/components/IntersectionsLayer/presence.js [hunk 5]
```diff
@@ -33,6 +33,12 @@ export class Presence {
     itineraries.forEach(({ author, waypoints }) => this.addOneItinerary(author, waypoints));
   }
 
+  /**
+   * Ads one itinerary to this Presence object.
```
Comment (blms) [inline src/components/IntersectionsLayer/presence.js:37]: Minor typo here ;)

**Review by blms — APPROVED** [review]
This is really excellent and thorough documentation, thank you so much. I really appreciate all the comments on the helper methods; even if they may seem self-explanatory, it's really helpful to orient oneself without having to think too much about code. The example shapes of return output are also very helpful.

I suggested a couple of minor tweaks; you can merge once those are completed.

**Comment by blms** [comment]
> P.S.: In parallel, I also started [writing a proper README fo Peripleo](https://github.com/peripleo/peripleo). I think I'll move some of this (specifically the list of UI components) into Wiki pages later. Either way: do let me know if you have feedback.

This is looking really great so far! My only suggestion is to include the imports in the example snippet. Otherwise I agree about moving the components into a dedicated documentation page/wiki. Very clear and well-written so far.

File: src/components/ITSBStore/itsbGraph.js (hunk 1 above)
Comment (rsimon) [reply src/components/ITSBStore/itsbGraph.js:143]: Ah - well, to my shame, I must admit I wasn't aware the rest parameter exists :-) I agree - that's really a nicer solution. I'm changing it.

File: src/components/ITSBStore/utils/likelihoodUtils.js (hunk 3 above)
Comment (rsimon) [reply src/components/ITSBStore/utils/likelihoodUtils.js]: Done!

**Comment by rsimon** [comment]
> My only suggestion is to include the imports in the example snippet. 

Good point - done!

**Pull request merged**